documentation.md

```diff
@@ -0,0 +1,669 @@
+
+_Read slots_ are defined identically, except that:
+
+- There is no _data_v_ field, since elementary burst read requests arrive simultaneously with the read address request.
+- There internal identifier for read and write address requests do not need to be of equal width, which depends on the maximal number of pending requests.
+
+The burst length does not need to be stored explicitly, as the _data_v_ and _mem_done_ bit arrays are set accordingly during the slot allocation.
+
+#### Delay estimation
+
+The delay estimation is performed using a decrementing counter (_rank_delay_cnt_) per rank (currently, only one rank is supported).
+When a simulated memory operation starts, the decrementing counter is set to a value corresponding to the situation:
+
+1. If this is a row hit.
+2. If this is a row miss, and there was no row in the buffer.
+3. If this is a row miss, and there was another row in the buffer.
+
+Rank states are represented by two additional elements (in addition to the decrementing counter):
+
+- _is_row_open_: Is set iff there is a row stored in the simulated row buffer.
+  Currently, this signal is only unset before the first memory request.
+- _row_start_addr_: Stores the identifier of the currently open row, if any.
+  The row identifier corresponds to all the most significant bits, that are not used for intra-row addressing.
+
+#### Release enable structures
+
+Write responses have a release enable structure made of one flip-flop per internal identifier.
+When the flip-flop is set, then the corresponding release eanble signal is propagated to the write response bank, which releases the signal if the requester is ready and if the signal is at the response head of its linked list.
+
+As read data come as a burst, and to allow the progressive release of the data in the burst, small counters are used to maintain the number of read data that can currently be released per internal identifier (_i.e._, per extended cell).
+The multi-hot release enable signal, similar to the one used for the write responses but with a possibly different width, has each bit set iff the corresponding counter is non-zero.
+
+When data is effectively released, the response banks notify this event using the _wrsp_released_iid_onehot_ and _rdata_released_iid_onehot_ one-hot signals.
+In the case of read data, the corresponding counter is decremented.
+In the case of write responses, the corresponding flip-flop is unset.
+
+<figure class="image">
+  <img src="https://i.imgur.com/zbh9VpO.png" alt="Delay calculator slots">
+  <figcaption>Fig: Delay calculator core: slots and release signals</figcaption>
+</figure>
+
+### Age management
+
+Age management is used in two point of the delay calculator core:
+
+- As many common scheduling strategies consider request relative age at some point, including the currently implemented strategy, the relative age between elementary burst entries must be (at least partially) maintained.
```

Comment:
- As many common scheduling strategies, including the currently implemented one, consider request relative age at some point, the relative age between elementary burst entries must be ....

Reply:
Done!